model/pipeline/src/main/proto/beam_runner_api.proto

```diff
@@ -500,6 +500,7 @@ message TestStreamPayload {
 
     message AdvanceWatermark {
       int64 new_watermark = 1;
+      string tag = 2;
```

Comment:
It will behave as expected: if unspecified then it will default to the Main PCollection Input. If specified then it will emit to that tagged PCollection. It should throw an exception if the tag doesn't exist. I added a comment to clarify the expected behavior.